### PR TITLE
Make CloudFunction sample to update the code within the same deployment

### PR DESCRIPTION
**Pull request by aljim**

### CODE REVIEW

File: examples/v2/cloud_functions/README.md [hunk 1]
```diff
@@ -4,28 +4,24 @@
...
-deploys a Cloud Function, a PubSub topic, and calls the cloud function.
+deploys a Cloud Function, and calls the cloud function.
 
 ## Deploy the template
 
-Use `cloud_function.yaml` to deploy this example template. If you would like to
-use your own cloud function, you need to upload the content to your GCS bucket,
-and change the sourceArchiveUrl parameter
+Use `cloud_function.yaml` to deploy this example template.
+This template use container builder to push the file contents into GCS so make
```
Comment (krwenholz) [inline examples/v2/cloud_functions/README.md]: s/use/uses
s/container builder/Container Builder

File: examples/v2/cloud_functions/README.md [hunk 2]
```diff
@@ -4,28 +4,24 @@
...
+sure you grant access to your container builder service account which is
+<projectNumber>@cloudbuild.gserviceaccount.com to be able to read your bucket
```
Comment (krwenholz) [inline examples/v2/cloud_functions/README.md]: Don't you mean write?

File: examples/v2/cloud_functions/README.md [hunk 3]
```diff
@@ -4,28 +4,24 @@
...
 
     gcloud deployment-manager deployments create my-function --config cloud_function.yaml
 
 ## Regarding the function source
 
-Our samples reference the function source (in the `function` directory) as
-`gs://cloud-function-sample/function.zip`. Cloud Functions doesn't have a
-way to inline the function code, so any function you write needs to be uploaded
-to a GCS bucket or use Cloud Source Repositories: instructions can be found
+Also is possible to use Cloud Source Repositories: instructions can be found
```
Comment (krwenholz) [inline examples/v2/cloud_functions/README.md]: s/is/it is/

File: examples/v2/cloud_functions/python/cloud_function.py [hunk 4]
```diff
@@ -11,54 +11,107 @@
...
-          function_call,
-      'action': ('gcp-types/cloudfunctions-v1beta2:'
-                 'cloudfunctions.projects.locations.functions.call'),
-      'properties': {
-          'name': '$(ref.' + function_name + '.name)',
-          'data': json.dumps({
-              'hola': 'mundo'
-          })
-      }
+  # Simulate a change on the function code by using a different import
```
Comment (krwenholz) [inline examples/v2/cloud_functions/python/cloud_function.py]: Is this necessary for any regular old cloud function? If so, we should probably make that clearer here. As it is, this seems like a fun line for testing or something.

Comment (aljim) [reply examples/v2/cloud_functions/python/cloud_function.py]: Woups comment was from another change that removed

**Comment by aljim** [comment]
Fixed comments

**Comment by aljim** [comment]
Done with the cleaning

File: examples/v2/cloud_functions/README.md [hunk 5]
```diff
@@ -4,28 +4,24 @@
 
 This is a [Google Cloud Deployment
 Manager](https://cloud.google.com/deployment-manager/overview) template that
-deploys a Cloud Function, a PubSub topic, and calls the cloud function.
+deploys a Cloud Function, and calls the cloud function.
```
Comment (krwenholz) [inline examples/v2/cloud_functions/README.md]: s/,//

File: examples/v2/cloud_functions/README.md [hunk 6]
```diff
@@ -4,28 +4,24 @@
...
 
 ## Deploy the template
 
-Use `cloud_function.yaml` to deploy this example template. If you would like to
-use your own cloud function, you need to upload the content to your GCS bucket,
-and change the sourceArchiveUrl parameter
+Use `cloud_function.yaml` to deploy this example template.
+This template uses container builder to push the file contents into GCS so make
```
Comment (krwenholz) [inline examples/v2/cloud_functions/README.md]: s/container builder/Container Builder

File: examples/v2/cloud_functions/python/cloud_function.py [hunk 7]
```diff
@@ -11,54 +11,106 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Creates a cloud function and calls it."""
 
+"""CloudFunction instance-type DM template.
+
+Provisions (or, creates) a pubsub topic.
```
Comment (krwenholz) [inline examples/v2/cloud_functions/python/cloud_function.py]: I thought we were not creating a pubsub topic?

File: examples/v2/cloud_functions/README.md (hunk 5 above)
Comment (krwenholz) [reply examples/v2/cloud_functions/README.md]: Never mind, I did it.

File: examples/v2/cloud_functions/README.md (hunk 6 above)
Comment (krwenholz) [reply examples/v2/cloud_functions/README.md]: Never mind, I did it.

File: examples/v2/cloud_functions/python/cloud_function.py (hunk 7 above)
Comment (krwenholz) [reply examples/v2/cloud_functions/python/cloud_function.py]: Never mind, I did it.

Comment (aljim) [reply examples/v2/cloud_functions/python/cloud_function.py]: Thank you :)

**Pull request merged**